tasks/Module 3/task_2.py

```diff
@@ -0,0 +1,57 @@
+"""Module2/task2: custom, python split function."""
+import re
+import pytest
+
+# Vars
+EMPTY_STRING_ERROR = 'Please provide a valid separator. ' \
+                     'It cannot be an empty string.'
+
+
+# Test data
+TEST_DATA_SET_1 = [
+    ('Python is cool', ' ', ['Python', 'is', 'cool']),
+    (',,,,', ',', ['', '', '', '', '']),
+    ('Python', 'Javascript', ['Python'])
+]
+TEST_DATA_SET_2 = [
+    ('feuer DPS pew pew', '')
+]
+
+
+def my_split(string_to_split, sep=None) -> list:
+    """Python's implementation of builtin (C) split function.
+
+    :param string_to_split: string that shall be splitted.
+    :param sep: separator by which string is splitted. None equals ' '
+    """
+    if sep == '':
+        raise ValueError(EMPTY_STRING_ERROR)
+
+    if sep is None:
+        sep = ' '
+        string_to_split = re.sub(r'[ \s]+', ' ', string_to_split)
```

Comment:
Regex can be just `\s+`. The special symbol `\s` matches any whitespace, so it includes space as well.

Reply:
Done.